docs/production/authentication-methods.md

```diff
@@ -337,10 +337,31 @@ More complex access control rules are possible via the
 2. If `org_membership` is not set or does not allow access,
    `AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL` will control access.
 
-This contains a map keyed by the organization's subdomain. The
-organization list with multiple maps, that contain a map with an attribute, and a required
-value for that attribute. If for any of the attribute maps, all user's
-LDAP attributes match what is configured, access is granted.
+`AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL` is a dictionary keyed by the organization's subdomain.
+The corresponding value is a list of `attribute: value` pair sets such that
+a user is permitted to access the organization if and only if
+the `attribute: value` pairs in at least one of these sets
+match the user's LDAP attributes.
+
+This is better illustrated with an example:
+
+```
+AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL = {
+    "zulip": [
+        {
+            "department": "main",
+            "employeeType": "staff"
+        },
+        {
+            "office": "Dallas"
+        }
+    ]
+}
+```
+
+This means that the organization `"zulip"` will be accessible via ldap authentication
```

Comment:
If I still understand my code (🙂) access to a realm not listed there is not granted if `AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL` is activated. This should probably be made a little bit more clear

Reply:
Oh, yeah you're right that should be explicit because otherwise it would be unexpected. Added this sentence:
```
... If this setting is enabled, organizations not explicitly
configured in this setting will not be accessible via ldap authentication at all.
```
and the end of the example:
```
This means that the organization `"zulip"` will be accessible via ldap authentication
only for users whose ldap attributes either contain both `department: main` `employeeType: staff`
or just `office: Dallas`. Other organizations will not be accessible, as they are not configured here.
```

Though I wonder if that's the behavior we really want? Or should organizations that don't get listed here simply have default access rather than becoming inaccessible through ldap?

Reply:
> Though I wonder if that's the behavior we really want? Or should organizations that don't get listed here simply have default access rather than becoming inaccessible through ldap?

I thought about that, too.
I do not know anymore why we decided to do it this way when implementing it.

Reply:
I think probably the theory was that you can arrange for an organization to have the "allow anything" policy by just having an empty list as your set of advanced policies?

I suspect this is probably aimed at avoiding typos resulting in configuration being inoperative, but I think I agree it'd be simpler to just have "If the string_id is not in `settings.AUTH_LDAP_ADVANCED_REALM_ACCESS_CONTROL`, then this setting doesn't apply to that subdomain".